ui/src/overridden/backoffice/Document/DocumentDetails/DocumentItems.js

```diff
@@ -0,0 +1,127 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { Link } from "react-router-dom";
+import {
+  BackOfficeRoutes,
+  itemApi,
+  ResultsTable,
+  Loader,
+  Error,
+  SeeAllButton,
+} from "@inveniosoftware/react-invenio-app-ils";
+import _get from "lodash/get";
+import _isEmpty from "lodash/isEmpty";
+
+export class DocumentItems extends Component {
+  componentDidMount() {
+    const { fetchDocumentItems, documentDetails } = this.props;
+    fetchDocumentItems(documentDetails.pid);
+  }
+
+  seeAllButton = () => {
+    const { documentDetails } = this.props;
+
+    const path = BackOfficeRoutes.itemsListWithQuery(
+      itemApi.query().withDocPid(documentDetails.pid).qs()
+    );
+    return <SeeAllButton to={path} />;
+  };
+
+  viewDetails = ({ row }) => {
+    return (
+      <Link
+        to={BackOfficeRoutes.itemDetailsFor(row.metadata.pid)}
+        data-test={row.metadata.pid}
+      >
+        {row.metadata.barcode}
+      </Link>
+    );
+  };
+
+  locationFormatter = ({ row }) => {
+    return `${row.metadata.internal_location.name} (${row.metadata.internal_location.location.name})`;
+  };
+
+  callNumberFormatter = ({ row }) => {
+    if (!_isEmpty(row.metadata.identifiers)) {
+      return row.metadata.identifiers.find(
+        (identifier) => identifier.scheme === "CALL_NUMBER"
+      ).value;
+    }
+    return "-";
+  };
+
+  renderTable(data) {
+    const { showMaxItems } = this.props;
+    const columns = [
+      {
+        title: "Barcode",
+        field: "metadata.barcode",
+        formatter: this.viewDetails,
+      },
+      { title: "Status", field: "metadata.status" },
+      { title: "Medium", field: "metadata.medium" },
+      {
+        title: "Location",
+        field: "metadata.internal_location.name",
+        formatter: this.locationFormatter,
+      },
+      {
+        title: "Call number",
+        field: "metadata.identifiers",
+        formatter: this.callNumberFormatter,
+      },
+      { title: "Restrictions", field: "metadata.circulation_restriction" },
+      {
+        title: "Loan Status",
+        field: "metadata.circulation.state",
+        formatter: ({ row, col }) => {
+          if (_get(row, col.field) === "ITEM_ON_LOAN") {
+            return (
+              <Link
+                to={BackOfficeRoutes.loanDetailsFor(row.metadata.circulation.loan_pid)}
+              >
+                on loan
+              </Link>
+            );
+          }
+          return _get(row, col.field) || "-";
+        },
+      },
+    ];
+    return (
+      <ResultsTable
+        data={data.hits}
+        columns={columns}
+        totalHitsCount={data.total}
+        name="attached items"
+        seeAllComponent={this.seeAllButton()}
+        showMaxRows={showMaxItems}
+      />
+    );
+  }
+
+  render() {
+    const { documentItems, isLoading, error } = this.props;
+    return (
+      <Loader isLoading={isLoading}>
```

Comment:
in invenio-app-ils you need to wrap this part with "DocumentItems.layout" which and pass this.renderTable as defined above. You will only override render function this way and avoid copying the whole component implementation, which is better for maintainability.
leave this part https://github.com/inveniosoftware/react-invenio-app-ils/pull/634/files#diff-18cb68f31fd99cfccf353466e41bb79106fcd071dfc552964bcfe7fa071a92f9R116 intact though, it is still a good improvement